src/SemaphoreAirdrops.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import { ERC20 } from 'solmate/tokens/ERC20.sol';
+import { ByteHasher } from './libraries/ByteHasher.sol';
+import { ISemaphore } from './interfaces/ISemaphore.sol';
+import { SafeTransferLib } from 'solmate/utils/SafeTransferLib.sol';
+
+/// @title Semaphore Multiple Airdrop Manager
+/// @author Miguel Piedrafita
+/// @notice Template contract for managing multiple airdrops using Semaphore groupds
+contract SemaphoreAirdrops {
+    using ByteHasher for bytes;
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  ERRORS                                ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Thrown when trying to create or update airdrop details without being the manager
+    error Unauthorized();
+
+    /// @notice Thrown when attempting to reuse a nullifier
+    error InvalidNullifier();
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  EVENTS                                ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Emitted when an airdrop is created
+    /// @param airdropId The id of the airdrop
+    /// @param airdrop The airdrop details
+    event AirdropCreated(uint256 airdropId, Airdrop airdrop);
+
+    /// @notice Emitted when an airdrop is successfully claimed
+    /// @param receiver The address that received the airdrop
+    event AirdropClaimed(uint256 indexed airdropId, address receiver);
+
+    /// @notice Emitted when the airdropped amount is changed
+    /// @param airdropId The id of the airdrop getting updated
+    /// @param airdrop The new details for the airdrop
+    event AirdropUpdated(uint256 indexed airdropId, Airdrop airdrop);
+
+    struct Airdrop {
```

Comment:
This is part of the public interface (through events and `getAirdrop`), so ideally it should be documented.